src/OmniSharp.MiscellanousFiles/OmniSharp.MiscellanousFiles.csproj

```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net461</TargetFramework>
+        <PlatformTarget>AnyCPU</PlatformTarget>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ProjectReference Include="..\OmniSharp.Abstractions\OmniSharp.Abstractions.csproj" />
+      <ProjectReference Include="..\OmniSharp.MSBuild\OmniSharp.MSBuild.csproj" />
```

Comment:
This shouldn't need to have a dependency on OmniSharp.MSBuild. Instead, I'd put a list of well-known project system names at a lower layer rather than referencing OmniSharp.MSBuild directly just to get the name of the project system.

Reply:
We need the project system type here, because we need to know when the project system finished initialization, before we can say that a particular document is a miscellanous document. So I have added this method - https://github.com/OmniSharp/omnisharp-roslyn/pull/1252/files#diff-63fe9836b6472be5b5ab510254cf5bb5R214 to the MSBuild project system and hence we need to know the `ProjectSystem` type for which we need the MSBuild reference.

Reply:
There are two possible approaches that we can take here:

1. Add `HasCompletedUpdateRequest` to the interface `IProjectSystem` interface
2. Make the GetWorkspaceModelAsync method public so that we can call it from the MiscellanousFiles system.

@DustinCampbell What do you think would be the best approach?

Reply:
I wouldn't go with option 2. Instead, I think we should work to break the coupling of the project systems, so the first option is the better approach.

Reply:
Another option might be to introduce another interface that can be used to check a project system for pending updates, implement it on the MSBuild project system, and then cast to the new interface in the miscellaneous files project system.

Reply:
@DustinCampbell Done. Added an IUpdates interface. Though I think we can have a better name for that. Any suggestion ?